pyteal/ast/txn.py

```diff
@@ -303,6 +303,13 @@ def first_valid(self) -> TxnExpr:
         """
         return self.makeTxnExpr(TxnField.first_valid)
 
+    def first_valid_time(self) -> TxnExpr:
+        """Get the UNIX timestamp of block before txn.FirstValid. Fails if negative
+        
+        For more information, see https://developer.algorand.org/docs/reference/transactions/#firstvalidtime
```

Comment:
Note: First valid time is not yet documented on the transactions reference page.